packages/react-storage/src/components/StorageBrowser/Views/Controls/Summary.tsx

```diff
@@ -1,38 +1,112 @@
 import React from 'react';
+import { withBaseElementProps } from '@aws-amplify/ui-react-core/elements';
+
+import type { OmitElements } from '../types';
 import { StorageBrowserElements } from '../../context/elements';
+import { CLASS_BASE } from '../constants';
 
-const { Button, Span, Text } = StorageBrowserElements;
+const { DefinitionDetail, DefinitionList, DefinitionTerm, View } =
+  StorageBrowserElements;
 
-interface Details<T extends StorageBrowserElements = StorageBrowserElements> {
-  (): React.JSX.Element;
-  Container: T['Span'];
-  Destination: T['Text'];
-  Completed: T['Text'];
-  Failed: T['Text'];
-  Canceled: T['Text'];
-  NotStarted: T['Text'];
+const BLOCK_NAME = `${CLASS_BASE}__details`;
+
+/* <Detail /> */
+const DetailContainer = withBaseElementProps(View, {
+  className: `${CLASS_BASE}__detail`,
+});
+
+const DetailLabel = withBaseElementProps(DefinitionTerm, {
+  className: `${CLASS_BASE}__detail__label`,
+});
+const DetailValue = withBaseElementProps(DefinitionDetail, {
+  className: `${CLASS_BASE}__detail__value`,
+});
+
+interface DetailProps {
+  label?: string;
+  value?: string;
+}
+interface Detail<T extends StorageBrowserElements = StorageBrowserElements> {
+  (props: DetailProps): React.JSX.Element;
+  Container: T['View'];
+  Label: T['DefinitionTerm'];
+  Value: T['DefinitionDetail'];
 }
 
-const Details: Details = () => <>Hi</>;
+const Detail: Detail = (props: DetailProps) => {
+  const { label, value } = props;
 
-Details.Canceled = Text;
-Details.Completed = Text;
-Details.Container = Span;
-Details.Destination = Text;
-Details.Failed = Text;
-Details.NotStarted = Text;
+  return (
+    <DetailContainer>
+      <DetailLabel>{label}</DetailLabel>
+      <DetailValue>{value}</DetailValue>
+    </DetailContainer>
+  );
+};
+Detail.Container = DetailContainer;
+Detail.Label = DetailLabel;
+Detail.Value = DetailValue;
 
-export interface SummaryControl<
+const CompletedControl = ({
+  label = 'Completed:',
+  value = '0/5',
+}: DetailProps) => <Detail label={label} value={value} />;
+
+const NotStartedControl = ({
+  label = 'Not started:',
+  value = '0/5',
+}: DetailProps) => <Detail label={label} value={value} />;
+
+const FailedControl = ({ label = 'Failed', value = '0/5' }: DetailProps) => (
+  <Detail label={label} value={value} />
+);
+
+const CanceledControl = ({
+  label = 'Canceled:',
+  value = '0/5',
+}: DetailProps) => <Detail label={label} value={value} />;
+
+const DestinationControl = ({
+  label = 'Destination:',
+  value = '/',
+}: DetailProps) => <Detail label={label} value={value} />;
+
+const Container = withBaseElementProps(DefinitionList, {
+  className: BLOCK_NAME,
+});
+
+interface _SummaryControl<
   T extends StorageBrowserElements = StorageBrowserElements,
 > {
   (): React.JSX.Element;
```

Comment:
Since the `Summary` is a list, can add `renderSummaryItem` here:
```
type RenderSummaryItem = (props: { label: string; value: string }) => React.JSX.Element;

interface _SummaryControl<
  T extends StorageBrowserElements = StorageBrowserElements,
> {
  (props: { renderSummaryItem?: RenderSummaryItem }): React.JSX.Element;
  ...
```

Reply:
Updated this in latest. I placed an example usage in SummaryControl
```
<Definition label="Completed" value="0/5" />
```